ql/src/java/org/apache/hadoop/hive/ql/ddl/function/AbstractFunctionAnalyzer.java

```diff
@@ -66,11 +70,24 @@ protected void addEntities(String functionName, String className, boolean isTemp
     }
     if (database != null) {
       outputs.add(new WriteEntity(database, WriteEntity.WriteType.DDL_NO_LOCK));
+      // Add the permanent function as a WriteEntity
+      Function function;
+      if (isCreate) {
```

Comment:
Can we use `queryState.getHiveOperation() == HiveOperation.CREATEFUNCTION` instead of passing an argument `isCreate`?

Reply:
Ack